File: library-base/src/main/java/com/dashingqi/base/application/ApplicationManager.java
package com.dashingqi.base.application;

import android.util.Log;

import com.dashingqi.base.providers.application.IApplicationProvider;
import com.orhanobut.logger.Logger;

import java.util.ArrayList;
import java.util.List;

/**
 * @author : zhangqi
 * @time : 2020/5/20
 * desc :
 */
class ApplicationManager {

    public static List<IApplicationProvider> apps = new ArrayList<>();

    static {
        Log.w("ApplicationManager", apps.toString());
    }

    public static void register(IApplicationProvider provider) {
        Log.w("ApplicationManager", provider.toString());
        apps.add(provider);
    }
}

File: library-base/src/main/java/com/dashingqi/base/application/ApplicationManager.kt
package com.dashingqi.base.application

import com.dashingqi.base.providers.application.IApplicationProvider

/**
 * @author : zhangqi
 * @time : 2020/5/20
 * desc :
 */
object ApplicationManager {
    /**
     * 用来存储实现IApplicationProvider接口的
     */
    @JvmStatic
    val apps = mutableListOf<IApplicationProvider>()

    /**
     * 用来注册实现IApplicationProvider的接口的类
     */
    @JvmStatic
    fun register(applicationProvider: IApplicationProvider) {
        apps.add(applicationProvider)
    }
}